File: packages/usf-ts/src/adapters/anthropic.ts
import type { CompiledPrompt } from "../prompt.js";

const DEFAULT_MODEL = "claude-sonnet-4-6";
const DEFAULT_MAX_TOKENS = 2048;

export function render(compiled: CompiledPrompt, opts: { model?: string } = {}): Record<string, unknown> {
  const fm = compiled.frontmatter;
  const hints = (fm["model_hints"] as Record<string, string> | undefined) ?? {};
  const model = opts.model ?? hints.anthropic ?? DEFAULT_MODEL;
  const payload: Record<string, unknown> = {
    model,
    system: compiled.system,
    messages: [{ role: "user", content: compiled.user }],
    max_tokens: Number(fm["max_tokens"] ?? DEFAULT_MAX_TOKENS),
  };
  if ("recommended_temperature" in fm) payload.temperature = fm["recommended_temperature"];
  return payload;
}

File: packages/usf-py/usf/adapters/anthropic.py
"""Anthropic Messages API adapter."""
from __future__ import annotations

from ..prompt import CompiledPrompt

DEFAULT_MODEL = "claude-sonnet-4-6"
DEFAULT_MAX_TOKENS = 2048


def render(compiled: CompiledPrompt, *, model: str | None = None) -> dict:
    fm = compiled.frontmatter
    chosen_model = model or (fm.get("model_hints") or {}).get("anthropic") or DEFAULT_MODEL
    payload: dict = {
        "model": chosen_model,
        "system": compiled.system,
        "messages": [{"role": "user", "content": compiled.user}],
        "max_tokens": int(fm.get("max_tokens", DEFAULT_MAX_TOKENS)),
    }
    if "recommended_temperature" in fm:
        payload["temperature"] = fm["recommended_temperature"]
    return payload
